ow_power_system/src/power_system_node.cpp

```diff
@@ -129,23 +79,69 @@ int main(int argc, char* argv[]) {
   std_msgs::Float64 soc_msg;
   std_msgs::Float64 rul_msg;
   ROS_INFO ("Power system node running");
-
-  int power_lifetime = power_csv.size();
+  
   while (ros::ok()) {
-    if (t_step < power_lifetime) {
-      soc_msg.data=power_csv[t_step][voltage_i];
-      rul_msg.data=power_csv[t_step][rul_i];
-    } //publishes last voltage value if sim time exceeds the length of the csv
-
-    //publish current SOC
-    SOC_pub.publish(soc_msg);
-    RUL_pub.publish(rul_msg);
-    t_step+=1;
-    
-    ros::spinOnce();
-    rate.sleep();
-   
-  }
 
+    // For each line of data in the example file, run a single prediction step.
+    for (const auto & line : power_data) {
+      // Get a new prediction
+      auto prediction = prognoser->step(line);
+      // Get the event for battery EoD. The first line of data is used to initialize the observer,
+      // so the first prediction won't have any events.
+      if (prediction.getEvents().size() == 0) {
+          continue;
+      }
+      auto eod_event = prediction.getEvents().front();
+      // The time of event is a `UData` structure, which represents a data
+      // point while maintaining uncertainty. For the MonteCarlo predictor
+      // used by this example, the uncertainty is captured by storing the
+      // result of each particle used in the prediction.
+      UData eod_time = eod_event.getTOE();
+      if (eod_time.uncertainty() != UType::Samples) {
+        std::cerr << "Unexpected uncertainty type for EoD prediction" << std::endl;
+        return 1;
```

Comment:
I understand this is part of their sample code but does there logic carries out to our use case? Why is error catastrophic and requires shutting down the power node? Can you elaborate?

Reply:
Very good points @Samahu. I think I rushed to get this PR in before the end of the day to show the functionality is there without tending to some of the other "cleanup" related issues created by the change. I should have taken the time to address these items beforehand. I was planning a second cleanup commit, but didn't anticipate your prompt review. Thank you

Reply:
This was an artifact of the example code that I had not yet cleaned up. The catastrophic nature of the error will be removed such that it is simply a ROS error printout

Reply:
Resolved.